grammar.js

```diff
@@ -1,157 +1,248 @@
+/* eslint-disable arrow-parens */
+/* eslint-disable-next-line spaced-comment */
+/// <reference types="tree-sitter-cli/dsl" />
+// @ts-check
+
+const primitives = [
```

Comment:
Do we gain anything from extracting `primitives` here?

From what I can see, it's only used in one place (cool trick with `...primitives`, btw, didn't know about it) - and in there it's just the same list.

Reply:
lots of other grammars use this style, and imo for anyone new to the language wanting to read through the grammar, having primitives at the top is a good start for anyone to gain some understanding about the primitive types

Reply:
Gotcha, I still don't see much value in it, but I don't feel strongly about it and am happy to follow the lead from other grammars.